### PR TITLE
Add tests to cover nil coercion #2897

### PR DESCRIPTION
**Pull request by suyograo**
Tests to cover https://github.com/elastic/logstash/issues/2897

### CODE REVIEW

File: spec/filters/grok_spec.rb [hunk 1]
```diff
@@ -702,4 +702,66 @@
...
+      }
+    CONFIG
+
+    sample "test 28.4ms" do
+      insist { subject["duration"] } == 28.4
+      insist { subject["tags"] }.nil?
+    end
+
+    sample "test N/A" do
+      insist { insist { subject.to_hash }.include?("duration") }.fails
```
Comment (jsvd) [inline spec/filters/grok_spec.rb]: maybe something like `insist { subject.include?("duration") } == false`?

Comment (suyograo) [inline spec/filters/grok_spec.rb]: will do

Comment (suyograo) [inline spec/filters/grok_spec.rb]: @jsvd actually, `include?` is buggy. I want to check if the key is in hash or not.. thats why the `to_hash`

Comment (jsvd) [inline spec/filters/grok_spec.rb]: ok, ship it, then :+1:

Comment (jordansissel) [inline spec/filters/grok_spec.rb]: `.fails` is for exceptions, basically asserting that any (I think?) exception is raised.

insist objects support predicate methods, so you can do this if you just want to check if a thing includes a thing:
```
insist { subject.to_hash }.include?("duration")
```

Some examples from insist's test suite on predicates:
* https://github.com/jordansissel/ruby-insist/blob/master/spec/insist/predicates_spec.rb
*

Comment (jordansissel) [inline spec/filters/grok_spec.rb]: One other note, the pattern of `insist { insist { } }.fails` is older and predates the `reject` method added later in insist's life.

Examples here use the old style: https://github.com/jordansissel/ruby-insist/blob/master/spec/insist/predicates_spec.rb#L48-L50

```
# Old style, assert something fails
insist { insist { foo }.empty? }.fails

# New style: rejecting success
reject { foo }.empty?
```

Comment (suyograo) [inline spec/filters/grok_spec.rb]: nice, i was looking for `reject`

Comment (jordansissel) [inline spec/filters/grok_spec.rb]: As a fun thing that I liked while designing insist, was that you could use `insist { ... }` as the subject:

```
subject do
  insist { something }
end

it "should be empty" do
  subject.empty?
end
```

It's not as explicit as I like, but I kind of like how it works. This pattern can be used with rspec's expect via `subject { expect(object) }` but due to expect's grammar pattern ("expect XYZ to ...") it doesn't read as well.

Some day we'll move the test suite fully away from insist and into using rspec expectations
